packages/xo-web/src/xo-app/new-vm/index.js

```diff
@@ -1065,16 +1069,25 @@ export default class NewVm extends BaseComponent {
     return name_label && template
   }
 
+  _onCpusChange = event => {
+    const value = getEventValue(event)
+    const cpusMax = Math.max(value, this.state.state.cpusMax)
+    this._setState({
+      CPUs: value,
+      cpusMax,
+    })
+  }
+
   _renderPerformances = () => {
-    const { coresPerSocket, CPUs, memoryDynamicMax } = this.state.state
+    const { coresPerSocket, CPUs, cpusMax, memoryDynamicMax } = this.state.state
     const { template } = this.props
     const { pool } = this.props
     const memoryThreshold = get(() => template.memory.static[0])
     const selectCoresPerSocket = (
       <SelectCoresPerSocket
-        disabled={pool === undefined}
+        disabled={pool === undefined || template === undefined}
```

Comment:
`template === undefined` should be enough.

Reply:
The cores per socket computation requires the `pool` property to be defined.

Reply:
But can `pool` be `undefined` if `template` isn't?

Reply:
yes in case of a self service user.

Reply:
Right!